drivers/1wire/1wire.c

```diff
@@ -668,7 +677,11 @@ int onewire_search(FAR struct onewire_master_s *master,
        */
     }
 
+  onewire_sem_post(master);
   return nslaves_match;
+err_unlock:
+  onewire_sem_post(master);
+  return ret;
```

Comment:
change to:
unlock:
  onewire_sem_post(master);
  return ret < 0 ? ret : nslaves_match;